app/components/repository-layout.js

```diff
@@ -12,12 +12,12 @@ export default Ember.Component.extend({
 
   @computed('repo.slug', 'repo.defaultBranch.name')
   statusImageUrl(slug, branchName) {
-    return this.get('statusImages').imageUrl(slug, branchName);
+    return this.get('statusImages').imageUrl(this.get('repo'), branchName);
   },
 
   @computed('repo.slug')
   urlGithub(slug) {
-    return this.get('externalLinks').githubRepo(slug);
+    return this.get('externalLinks').githubRepo(this.get('repo'));
```

Comment:
These could both have `repo` as their first dependent keys now, instead of `slug`, right?

Reply:
I actually want to observe `repo.slug`, because this computed property should recompute when slug changes (otherwise it would recompute only when repo changes to another repo). But that got me thinking that this code is not really readable. I pass the entire repo, because I need `slug` and `private` properties (and then it means I also need to depend on `private`. So to make this less confusing I'll change it to:

```js
@computed('repo.slug', 'repo.private')
   urlGithub(slug, private) {
    return this.get('externalLinks').githubRepo(slug, private);
  }
```

(this is why I hate the fact that Ember.js requires listing dependencies, btw ;) ).

Reply:
oh yes, I forgot about that reason about what to observe. Well, nice change, thanks.